src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/GetDocumentByResourceName.cs

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using EdFi.DataManagementService.Core.External.Backend;
+using Microsoft.Extensions.Logging;
+using Npgsql;
+
+namespace EdFi.DataManagementService.Backend.Postgresql.Operation;
+
+public interface IGetDocumentByResourceName
+{
+    public Task<GetResult> GetByResourceName(IGetRequest getRequest, int offset, int limit);
+}
+
+public class GetDocumentByResourceName(NpgsqlDataSource _dataSource, ILogger<GetDocumentByResourceName> _logger) : IGetDocumentByResourceName
+{
+    public async Task<GetResult> GetByResourceName(IGetRequest getRequest, int offset, int limit)
+    {
+        _logger.LogDebug("Entering GetDocumentByResourceName.GetByResourceName - {TraceId}", getRequest.TraceId);
+        try
+        {
+            await using var connection = await _dataSource.OpenConnectionAsync();
+
+            await using NpgsqlCommand command =
```

Comment:
Please move the usage of NpgsqlCommand into `SqlAction`. Update and upsert are both doing this, but we haven't moved get-by-id yet. The comment on `SqlAction` explains the division of responsibility with the action classes.